### ISSUE TITLE
The front-matter is not parsed if the tags are in YAML 

### ISSUE THREAD

**Issue by arulrajnet** [jonathan-s/pelican-obsidian#12]
## Issue

This is the follow-up issue of #5 

The front-matter is as like below. The `tags` is in YAML format, and it's in between of other metadata. 

```
---
title: No More Basic Auth &#58; htpasswd with OAuth2 Proxy
date: 2024-12-15 20:44:14
author: arul
category: Security
tags:
  - proxy
  - Traefik
  - APISix
  - nginx
slug: no-more-basic-auth-htpasswd-with-oauth2-proxy
cover: /assets/images/basic-auth-oauth2-proxy-cover.png
color: gray
headline: Create cookie session store for htpasswd based authentication using oauth2-proxy. This will eliminate the insecure basic authentication.
status: published
---
```

Whatever metadata after `tags` is not parsing. This article automatically moved to drafts, since the published metadata is at the end. 

## Workaround

Now I have changed the order and placed the `tags` at the end. 

```
---
title: No More Basic Auth &#58; htpasswd with OAuth2 Proxy
date: 2024-12-15 20:44:14
author: arul
category: Security
slug: no-more-basic-auth-htpasswd-with-oauth2-proxy
cover: /assets/images/basic-auth-oauth2-proxy-cover.png
color: gray
headline: Create cookie session store for htpasswd based authentication using oauth2-proxy. This will eliminate the insecure basic authentication.
status: published
tags:
  - proxy
  - Traefik
  - APISix
  - nginx
---
```
Now it's working as expected. 

## What Next?

Pelican talks about plugin [ordering](https://docs.getpelican.com/en/latest/plugins.html#how-to-create-plugins) in `PLUGINS`. But it's not working. 

Maybe the metadata received to this plugin without others. 

FYI, The unit test is passing. Because it takes care about only `ObsidianMarkdownReader`

**Comment by jonathan-s** [comment]
I suspect that this will be solved by #11 as we solve this in its entirety.

**Comment by arulrajnet** [comment]
@jonathan-s That tag issue fixed. 

But now the images are not loaded. 

![image](https://github.com/user-attachments/assets/c9cd662a-fb26-4114-a6d5-7d9a7253d5c5)


Here is the example post https://attila.demo.arulraj.net/2015/11/pelican-obsidian-example#images

The code is in https://github.com/arulrajnet/attila-demo/blob/master/content/pelican-obsidian-example.md

Follow the [steps](https://github.com/arulrajnet/attila-demo/blob/master/README.md#clone-and-install-theme) 

```
git clone --depth=1 https://github.com/arulrajnet/attila-demo
cd attila-demo
git submodule init
cd pelican-plugins
git pull
git checkout master
cd ..
virtualenv .venv
source .venv/bin/activate
pip install .
```

Then

```
make html
```

Then

```
make serve
```
Now check http://localhost:8000/2015/11/pelican-obsidian-example#images
